grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1490,12 +1494,12 @@ class StockMovementService {
                     def picklistItems = requisitionItem.substitutionItems.findAll { it.product == associatedProduct }
                             .collect { it.picklistItems }?.flatten()
 
-                    availableItems = productAvailabilityService.getAllAvailableBinLocations(location, associatedProduct)
+                    availableItems = productAvailabilityService.getAllAvailableBinLocations(location, associatedProduct?.id)
                     availableItems = availableItems.findAll { it.quantityOnHand > 0 }
```

Comment:
@kchelstowski one more thing. I think you can also use here the same logic with `excludeNegativeQuantity` parm and pass it down to the `getAvailableItems` (since it is false by default)

Reply:
What should be enough here is to just change the method from `getAllAvailableBinLocations` to `getAvailableBinLocations`, since the only different between them is that flag, that makes this filtering:
```groovy
List<AvailableItem> getAvailableBinLocations(Location location, List<String> productsIds) {
        List<AvailableItem> availableItems = getAvailableItems(location, productsIds, true)

        availableItems = sortAvailableItems(availableItems)

        return availableItems
}

// Include also bin locations with negative qty (needed for edit page items)
List<AvailableItem> getAllAvailableBinLocations(Location location, List products) {
  List<AvailableItem> availableBinLocations = getAvailableItems(location, products)
  return availableBinLocations
}
```
so I'd say we should not have used the `getAll` method here if we filter the results below.